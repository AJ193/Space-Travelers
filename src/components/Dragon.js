import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getDragons, updateDragon } from '../redux/dragons/dragonsSlice';

const Dragon = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getDragons());
  }, [dispatch]);

  const { dragons, loading } = useSelector((state) => state.dragons);

  const onUpdate = (id) => {
    dispatch(updateDragon(id));
  };

  const onCancel = (id) => {
    dispatch(cancelDragon(id));
  };
  

  if (loading) return <p>Loading...</p>;

  return (
    <div className="container mt-5 dragon">
      <div className="list">
        {dragons.map((dragon) => (
          <div key={dragon.id} className="row mb-5">
            <img className="col-md-3 image" src={dragon.flickr_images} alt="falcon1" />
            <div className="col-md-9">
              <h2 className="title">{dragon.name}</h2>
              <p className="title">{dragon.type}</p>
              {
                dragon.reserved ? <button onClick={() => onCancel(dragon.id)} type="button" className="btn btn-outline-primary">Cancel Reservation</button>
                  : <button onClick={() => onUpdate(dragon.id)} type="button" className="btn btn-primary">Reserve Rocket</button>
              }

            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Dragon;
