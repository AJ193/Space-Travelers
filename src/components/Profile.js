import React from 'react';
import { useSelector } from 'react-redux';
import { selectRockets } from '../redux/rocket/rocketsSlice';

const Profile = () => {
  const rocketsList = useSelector(selectRockets);
  const { missionItem } = useSelector((store) => store.missions);
  const { dragons } = useSelector((store) => store.dragons);

  const reservedDragon = dragons.filter((dragon) => dragon.reserved === true);
  const reservedMissions = missionItem.filter((mission) => mission.reserved === true);
  const reservedRockets = rocketsList.filter((b) => b.reserved);
  const generateKey = () => `_${Math.random().toString(36).substr(2, 9)}`;

  return (
    <section>
      <div className="container-mt-5">
        <div className="row">
          <div className="col-4">
            <h2>Missions</h2>
            <table className="table table-bordered">
              { reservedMissions.map((mission) => (
                <tr key={mission.mission_id}><td className="p-2">{ mission.mission_name }</td></tr>
              ))}
            </table>
          </div>
          <div className="col-4">
            <h1 className="Rockets">Rockets</h1>
            <ul>
              {reservedRockets.map((rocket) => (
                <li key={generateKey()}>{rocket.rocket_name}</li>
              ))}
            </ul>
          </div>
        </div>
        <div className="col-4">
          <h2>Dragons</h2>
          <table className="table table-bordered">
            { reservedDragon.map((dragon) => (
              <tr key={dragon.id}><td className="p-2">{ dragon.name }</td></tr>
            ))}
          </table>
        </div>
      </div>
    </section>
  );
};

export default Profile;
