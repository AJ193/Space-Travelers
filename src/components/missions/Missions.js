import React from 'react';
import { useSelector } from 'react-redux';
import MissionItem from './MissionItem';

const Missions = () => {
  const { missionItem, isLoading } = useSelector((store) => store.missions);

  if (isLoading) {
    return (
      <div className="loader-container">
        Loading...
      </div>
    );
  }

  return (
    <div className="container-fluid mt-5">
      <table className="table table-striped table-bordered">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th aria-label=""> </th>
          </tr>
        </thead>
        <tbody>
          {missionItem.map((mission) => (
            <MissionItem
              key={mission.mission_id}
              id={mission.mission_id}
              name={mission.mission_name}
              description={mission.description}
              reserved={mission.reserved || false}
            />
          ))}
        </tbody>

      </table>
    </div>
  );
};

export default Missions;
