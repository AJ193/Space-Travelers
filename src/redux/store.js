import { configureStore } from '@reduxjs/toolkit';
import dragonReducer from './dragons/dragonsSlice';
import missionReducer from './missions/missionSlice';
import rocketsReducer from './rocket/rocketsSlice';

const store = configureStore({
  reducer: {
    dragons: dragonReducer,
    missions: missionReducer,
    rockets: rocketsReducer,
  },
});

export default store;
