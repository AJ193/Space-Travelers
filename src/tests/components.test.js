import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../components/missions/Missions';
import Rockets from '../components/Rockets';
import Profile from '../components/Profile';
import Dragon from '../components/Dragon';

describe('Component testing', () => {
  test('Mission component should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Missions />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Rocket component should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Rockets />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Profile component should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Dragon component should render correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Dragon />
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
