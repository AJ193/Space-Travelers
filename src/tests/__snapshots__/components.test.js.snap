// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing Dragon component should render correctly 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;

exports[`Component testing Mission component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container-fluid mt-5"
  >
    <table
      class="table table-striped table-bordered"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th
            aria-label=""
          >
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;

exports[`Component testing Profile component should render correctly 1`] = `
<DocumentFragment>
  <section>
    <div
      class="container mt-5"
    >
      <div
        class="row"
      >
        <div
          class="col-4"
        >
          <h2>
            Missions
          </h2>
          <table
            class="table table-bordered"
          >
            <tbody />
          </table>
        </div>
        <div
          class="col-4"
        >
          <h2
            class="Rockets"
          >
            Rockets
          </h2>
          <table
            class="table table-bordered"
          >
            <tbody />
          </table>
        </div>
        <div
          class="col-4"
        >
          <h2>
            Dragons
          </h2>
          <table
            class="table table-bordered"
          >
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Component testing Rocket component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <ul
      class="rockets-container"
    />
  </div>
</DocumentFragment>
`;
